html/browsers/browsing-the-web/navigating-across-documents/initial-empty-document/window-open-204-fragment.html

```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
+<meta charset="UTF-8">
+<title>Fragment navigation on initial empty document created through window.open(204-url)</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/helpers.js"></script>
+<body></body>
+<script>
+"use strict";
+const postMessageToOpenerOnLoad = `
+    window.onload = () => {
+      window.opener.postMessage("loaded", "*")
+    }
+`;
+const url1 = "#foo";
+const url2 = "resources/code-injector.html?2&pipe=sub(none)&code=" + encodeURIComponent(postMessageToOpenerOnLoad);
+
+promise_test(async t => {
+  // Open a new window with a URL that doesn't load a new document, so it will stay in the initial empty document.
+  const openedWindow = windowOpen204(t);
+
+  // Do a fragment navigation within the initial empty document through setting location.href.
+  openedWindow.location.href = url1;
```

Comment:
I wonder if this test is breaking because it's parsing `#foo` relative to this document, and not relative to 204-url.

Try `openedWindow.location.hash = "foo"` instead?

Reply:
I tried that and it seems to fail a lot more (then again I changed a lot of other parts as well). Do you think we should use location.hash now?

Reply:
I'm just worried that this is trying to load `window-open-204-fragment.html#foo` instead of `/common/blank.html?pipe=status(204)#foo`. Maybe just explicitly do `openedWindow.location.href = "/common/blank.html?pipe=status(204)#foo"`?

Reply:
Oh, I think that's one of the source of flakiness. I changed it to navigate to `about:blank#foo` explicitly now!